**Changes Requested**

The implementation approach is sound -- creating virtual Pad objects at escape endpoint coordinates so the main routing pipeline (RSMT + A*) routes to escape endpoints instead of original pad centers is a clean design. The code is well-commented with issue references, and the new test file covers the key behaviors. However, there is a blocking test failure that must be fixed before approval.

## Blocking Issue

**Failing test: `tests/test_global_router.py::TestEscapeRouteRipupEligibility::test_route_with_escape_passes_initial_routes_to_two_phase`**

This test (from Issue #2294) asserts that `initial_routes=escape_routes` is passed to `route_all_two_phase()`. The PR intentionally removes this parameter (escape routes are now permanent infrastructure, not rip-up-eligible initial routes), but the existing test was not updated. It currently fails with:

```
AssertionError: assert 'initial_routes' in {'corridor_width_factor': 2.0, 'progress_callback': None, 'timeout': None, 'use_negotiated': True}
```

This test needs to be either:
1. **Updated** to assert that `initial_routes` is NOT passed (reflecting the new design), or
2. **Replaced** with a test that verifies escape routes are preserved as permanent infrastructure (e.g., checking that escape segments remain on the grid after routing)

## Minor Observation (Non-blocking)

`_escape_protected_nets` is populated during `generate_escape_routes()` but is never read by any production code path. The PR description says escape stubs should "not be ripped up," and a test verifies the set is populated, but no routing code actually checks this set to skip rip-up. This appears to be preparatory infrastructure for a future change. Consider either:
- Adding the rip-up protection logic that reads `_escape_protected_nets`, or
- Adding a code comment noting this is reserved for future use

## Code Quality Notes (Positive)

- Consistent application of the override pattern across all 7 routing code paths
- Good use of `dict.get(key, fallback)` for clean mixed-pad resolution
- Virtual pad construction correctly preserves net, ref, pin, and uses `escape.escape_layer`
- New tests cover virtual pad creation, RSMT integration, mixed pads, layer preservation, and protected nets
- Comments are clear and reference the issue number throughout

## Test Scoping

**Strategy**: Direct pytest execution in worktree
**Result**: 5/5 new tests pass; 132/133 escape-related tests pass (1 pre-existing test fails due to the design change described above)

--- a/src/kicad_tools/router/core.py
+++ b/src/kicad_tools/router/core.py
@@ -457,9 +457,6 @@ class Autorouter:
         # escape stub segments.
         # Maps (ref, pin) -> virtual Pad at escape endpoint.
         self._escape_pad_overrides: dict[tuple[str, str], Pad] = {}
-        # Nets whose escape route segments should not be ripped up.
-        self._escape_protected_nets: set[int] = set()
-
         # Fine-grid routing count (updated by route_all_multi_resolution)
         self.fine_grid_nets_count: int = 0
 
@@ -5419,7 +5416,6 @@ class Autorouter:
                         drill=pad.drill,
                     )
                     self._escape_pad_overrides[pad_key] = virtual_pad
-                    self._escape_protected_nets.add(pad.net)
 
             print(
                 f"  Escape routes: {package.ref} ({package.package_type.name})"

--- a/tests/test_escape_endpoint_routing.py
+++ b/tests/test_escape_endpoint_routing.py
@@ -138,59 +138,6 @@ class TestEscapePadOverrides:
             assert virtual_pad.ref == original_pad.ref
             assert virtual_pad.pin == original_pad.pin
 
-    def test_escape_protected_nets_populated(self):
-        """Nets with escape routes are added to _escape_protected_nets."""
-        from kicad_tools.router.core import Autorouter
-
-        rules = DesignRules(
-            trace_width=0.2,
-            trace_clearance=0.2,
-            via_drill=0.3,
-            via_diameter=0.6,
-        )
-
-        router = Autorouter(width=30.0, height=30.0, rules=rules)
-
-        pads = []
-        net = 1
-        for i in range(10):
-            pads.append({
-                "x": 10.0 + i * 0.65,
-                "y": 13.0,
-                "width": 0.3,
-                "height": 0.8,
-                "net": net,
-                "net_name": f"NET_{net}",
-                "layer": Layer.F_CU,
-                "number": str(i + 1),
-            })
-            net += 1
-        for i in range(10):
-            pads.append({
-                "x": 10.0 + i * 0.65,
-                "y": 17.0,
-                "width": 0.3,
-                "height": 0.8,
-                "net": net,
-                "net_name": f"NET_{net}",
-                "layer": Layer.F_CU,
-                "number": str(i + 11),
-            })
-            net += 1
-
-        router.add_component("U1", pads)
-
-        dense = router.detect_dense_packages()
-        if not dense:
-            pytest.skip("Package not detected as dense")
-
-        router.generate_escape_routes(dense)
-
-        assert len(router._escape_protected_nets) > 0, (
-            "Expected escape protected nets to be populated"
-        )
-
-
 class TestRSMTUsesEscapeEndpoints:
     """Test that build_rsmt uses virtual pad positions."""
 

--- a/tests/test_global_router.py
+++ b/tests/test_global_router.py
@@ -1450,8 +1450,9 @@ class TestEscapeRouteRipupEligibility:
             autorouter.route_with_escape()
             mock_prepass.assert_called_once()
 
-    def test_route_with_escape_passes_initial_routes_to_two_phase(self, autorouter):
-        """Escape routes are passed as initial_routes to route_all_two_phase."""
+    def test_route_with_escape_does_not_pass_initial_routes(self, autorouter):
+        """Escape routes are NOT passed as initial_routes; main routing connects
+        to escape endpoints via virtual pad overrides instead."""
         from unittest.mock import MagicMock, patch
 
         autorouter.add_component(
@@ -1479,8 +1480,7 @@ class TestEscapeRouteRipupEligibility:
             autorouter.route_with_escape()
             mock_two_phase.assert_called_once()
             call_kwargs = mock_two_phase.call_args[1]
-            assert "initial_routes" in call_kwargs
-            assert call_kwargs["initial_routes"] == [fake_escape]
+            assert "initial_routes" not in call_kwargs
 
     def test_two_phase_initial_routes_seeded_into_net_routes(self, rules):
         """Initial routes are seeded into the negotiated router's net_routes."""

